File: reactive.infra.form.engine/src/main/java/org/infra/reactive/form/engine/form/engine/model/dto/request/common/PagingDTO.java
package org.infra.reactive.form.engine.form.engine.model.dto.request.common;

import lombok.Data;

@Data
public class PagingDTO {
    private long totalRecord;
    private long fromRecord;
    private long toRecord;

    public long getPageSize() {
        return toRecord - fromRecord;
    }
}

File: reactive.infra.web/src/main/resources/static/Resources/JS/ERPFrameWork/Communicate/Models/Request/Common/PagingDTO.js
import {BaseModel} from "../../../../../UIFrameWork/Shared/Common/BaseModel.js";

export default class PagingDTO extends BaseModel {
    constructor() {
        super();
    }

    setTotalRecord(totalRecord) {
        this.totalRecord = totalRecord;
    }

    setFromRecord(fromRecord) {
        this.fromRecord = fromRecord;
    }

    setToRecord(toRecord) {
        this.toRecord = toRecord;
    }

    getFromRecord() {
        return this.fromRecord;
    }

    getToRecord() {
        return this.toRecord;
    }

    getTotalRecord() {
        return this.totalRecord;
    }

    getPageSize() {
        return this.toRecord - this.fromRecord;
    }

    getAllPage() {
        return Math.round(this.totalRecord / this.getPageSize());
    }

    getPageNumber() {
        return Math.round(this.fromRecord / this.getPageSize()) + 1;
    }
}